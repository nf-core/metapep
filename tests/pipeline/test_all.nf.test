nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_all"
    tag "pipeline"

    test("test all input path including protein downloads") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },

                { assert snapshot(path("$outputDir/db_tables/alleles.tsv"),
                                path("$outputDir/db_tables/conditions_alleles.tsv"),
                                path("$outputDir/db_tables/conditions.tsv"),
                                path("$outputDir/db_tables/entities_proteins.tsv"),
                                path("$outputDir/db_tables/entities.tsv"),
                                path("$outputDir/db_tables/microbiomes_entities.no_weights.tsv"),
                                path("$outputDir/db_tables/microbiomes_entities.nucl.tsv"),
                                path("$outputDir/db_tables/microbiomes_entities.tsv"),
                                path("$outputDir/db_tables/microbiomes.tsv"),
                                path("$outputDir/db_tables/proteins_peptides.tsv"),
                                path("$outputDir/db_tables/peptides.tsv.gz"),
                                path("$outputDir/db_tables/proteins.tsv.gz"),
                                path("$outputDir/db_tables/predictions.tsv.gz"),
                                path("$outputDir/db_tables/stats.txt")).match("db_tables") },

                { assert snapshot(path("$outputDir/entrez_data/microbiomes_entities.entrez.tsv")).match("entrez_data") },
                { assert new File("$outputDir/entrez_data/entities_proteins.entrez.tsv").exists() },
                { assert new File("$outputDir/entrez_data/proteins.entrez.tsv.gz").exists() },
                { assert new File("$outputDir/entrez_data/taxa_assemblies.tsv").exists() },

                { assert new File("$outputDir/figures/entity_binding_ratios.A_01_01.pdf").exists() },
                { assert new File("$outputDir/figures/entity_binding_ratios.with_points.A_01_01.pdf").exists() },
                { assert new File("$outputDir/figures/prediction_score_distribution.A_01_01.pdf").exists() },

                { assert snapshot(path("$outputDir/logs/prediction_warnings.log"),
                                path("$outputDir/logs/SUCCEEDED_unify_peptide_lengths.log")).match("logs") },
                { assert new File("$outputDir/logs/download_proteins.log").exists() },

                { assert new File("$outputDir/prodigal/proteins.pred_1.bin.1.tsv.gz").exists() },
                { assert new File("$outputDir/prodigal/proteins.pred_1.bin.2.tsv.gz").exists() },
                { assert new File("$outputDir/prodigal/proteins.pred_2.tsv.gz").exists() }
            )

        }

    }

}
