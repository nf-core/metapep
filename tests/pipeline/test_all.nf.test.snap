{
    "db_tables": {
        "content": [
            "alleles.tsv:md5,77994270b1eab3e8363c90d8d09780fa",
            "conditions_alleles.tsv:md5,0e992c19471bb56c9b6e6a6f71302742",
            "conditions.tsv:md5,87707cd0dee14b0ca4face090ba9dce9",
            "entities.tsv:md5,d8fb481a73a1c92298ce2b093ec54742",
            "microbiomes_entities.nucl.tsv:md5,aa0d4cc3546508c8e5b81bc9b4b16e39",
            "microbiomes.tsv:md5,70871c8b3023f5a8412c05f4a69dda2f",
            "stats.txt:md5,896155e0f043f30bd71c38c8ee7b466c"
        ],
        "timestamp": "2024-06-26T08:14:54+0000"
    },
    "entrez_data": {
        "content": [
            "microbiomes_entities.entrez.tsv:md5,77463a684452d34795b6eda52cfad072"
        ],
        "timestamp": "2024-06-26T08:14:54+0000"
    },
    "logs": {
        "content": [
            "prediction_warnings.log:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "timestamp": "2024-06-26T08:14:54+0000"
    }
}
